tests/Issues/EmptyBooleanCompare/Fixture/array_dim_fetch_from_docblock.php.inc

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Rector\Core\Tests\Issues\EmptyBooleanCompare\Fixture;
+
+final class ArrayDimFetchFromDocblock
+{
+    /**
+     * @param array{host: string} $parts
+     */
+    public function checkUrl(array $parts)
```

Comment:
@staabm the specific type of `$parts['host']` from docblock will be detected as host: `StringType` instead of `MixedType`

that can be feature or bug :)

Reply:
if the shape was returned by a native extension, I would say its a feature.

if the shape is declared as a phpdoc, I would say its a bug :)

Reply:
how to detect if it is from native extension ?

Reply:
I think we can't detect it. but maybe we can enhance our own `getNativeType()` method to apply the phpstan-src native extensions (if thats not the case already)